### ISSUE TITLE
get_image() gives wrong result on GradientTexture2D 

### ISSUE THREAD

**Issue by bionick7** [godotengine/godot#66627]
### Godot version

4.0 beta2 (mono version)

### System information

Ubuntu 22

### Issue description

the get_image() command gets an image with both wrong size and wrong general look when called on GradientTexture2D.
It seams the the image is  always 64x64 pixels. It depicts a different gradient with generally the same colors.

### Steps to reproduce

Put the following code on a Sprite2D and put any gradient not 64x64 pixels in size into "gradient_texture"

```gdscript
extends Sprite2D

@export var gradient_texture: GradientTexture2D 

func _ready():
	var img = gradient_texture.get_image()
	prints(img.get_width(), img.get_height())
	texture = ImageTexture.create_from_image(img)
```

### Minimal reproduction project

[GradientTexture2DBug.zip](https://github.com/godotengine/godot/files/9679333/GradientTexture2DBug.zip)

**Comment by timothyqiu** [comment]
Size properties are set after `gradient`:

https://github.com/godotengine/godot/blob/56563ef09e779b9f4b105fea0e5a1a8e59c7cfe8/scene/resources/texture.cpp#L2518-L2520

`set_gradient()` updates the image directly, while `set_width()` and `set_height()` defers the update:

https://github.com/godotengine/godot/blob/56563ef09e779b9f4b105fea0e5a1a8e59c7cfe8/scene/resources/texture.cpp#L2414-L2418

So in `_ready()`, the image you get still uses the default 64x64 size, and there's a size update pending.

As a workaround, you can put either snippet at the beginning of `_ready()`:

```
# Wait for one frame
await get_tree().process_frame

# Or something like this to trigger the image update
var g := gradient_texture.gradient
gradient_texture.gradient = null
gradient_texture.gradient = g
```

**Comment by Calinou** [comment]
The update is likely deferred so that the gradient isn't generated twice when both width and height are set at the same time.

**Comment by knil79** [comment]
This is confirmed in 4.0.2 as well, is there already a commit for the documentation update with the work around?

**Comment by KoBeWi** [comment]
Fixed by #81137
